packages/react/src/components/ContentSwitcher/ContentSwitcher.js

```diff
@@ -5,125 +5,135 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import PropTypes from 'prop-types';
-import React from 'react';
-import classNames from 'classnames';
 import { settings } from 'carbon-components';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React, {
+  Children,
+  cloneElement,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { composeEventHandlers } from '../../tools/events';
 import { getNextIndex, matches, keys } from '../../internal/keyboard';
 
 const { prefix } = settings;
 
-export default class ContentSwitcher extends React.Component {
-  /**
-   * The DOM references of child `<Switch>`.
-   * @type {Array<Element>}
-   * @private
-   */
-  _switchRefs = [];
-
-  state = {};
-
-  static propTypes = {
-    /**
-     * Pass in Switch components to be rendered in the ContentSwitcher
-     */
-    children: PropTypes.node,
-
-    /**
-     * Specify an optional className to be added to the container node
-     */
-    className: PropTypes.string,
-
-    /**
-     * Specify an `onChange` handler that is called whenever the ContentSwitcher
-     * changes which item is selected
-     */
-    onChange: PropTypes.func.isRequired,
-
-    /**
-     * Specify a selected index for the initially selected content
-     */
-    selectedIndex: PropTypes.number,
-  };
-
-  static defaultProps = {
-    selectedIndex: 0,
-  };
-
-  static getDerivedStateFromProps({ selectedIndex }, state) {
-    const { prevSelectedIndex } = state;
-    return prevSelectedIndex === selectedIndex
-      ? null
-      : {
-          selectedIndex,
-          prevSelectedIndex: selectedIndex,
-        };
+function ContentSwitcher({
+  children,
+  className: customClassName,
+  onChange,
+  selectedIndex: controlledSelectedIndex = 0,
+  ...rest
+}) {
+  const switchRefs = [];
+  const className = cx(`${prefix}--content-switcher`, customClassName);
+  const savedOnChange = useRef(null);
+  const [selectedIndex, setSelectedIndex] = useState(controlledSelectedIndex);
+  const [prevControlledIndex, setPrevControlledIndex] = useState(
+    controlledSelectedIndex
+  );
+  const [shouldFocus, setShouldFocus] = useState(false);
+
+  if (controlledSelectedIndex !== prevControlledIndex) {
+    setSelectedIndex(controlledSelectedIndex);
+    setPrevControlledIndex(controlledSelectedIndex);
+    setShouldFocus(false);
```

Comment:
Does the rendering code after this line stop focusing or not?

Reply:
Here's a quick isolated demo that I think demonstrates what's going on here: https://codesandbox.io/s/jolly-antonelli-uevex

Associated demo:

![demo](https://user-images.githubusercontent.com/3901764/61137124-88dae600-a48a-11e9-8ef4-0379be0aad5d.gif)

All this is to say that when we enqueue this state transition the effect is never re-run with the truth-y value in the effect. Instead, React runs the effect with the false-y value.

Reply:
When you view the storybook, you'll notice that on first render we have no focus inside the ContentSwitcher alongside when selectedIndex is changed through the knob 👍 

![demo](https://user-images.githubusercontent.com/3901764/61137373-00107a00-a48b-11e9-88ae-d28d42ae2f95.gif)

Reply:
Cool, makes sense now, thank you for clarifying.